_posts/2025-06-18-#31829.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: pr
+date: 2025-06-18
+title: "Improve TxOrphanage denial of service bounds"
+pr: 31829
+authors: [glozow]
+components: ["p2p"]
+host: glozow
+status: upcoming
+commit: 0d511965c91
+---
+
+## Notes
+
+- The [PR description](https://github.com/bitcoin/bitcoin/pull/31829#issue-2840961349) summarizes the motivations for
+  this change and the new eviction strategy: we must ensure the orphanage is DoS-resistant, but also want to prevent any
+peer from affecting another peer's usage of orphanage space.
+
+- The PR does a few things: virtualizes the `TxOrphanage` class inherited by `TxOrphanageImpl`, implements a new
+  eviction strategy, rewrites `TxOrphanage` the data structure(s) as a single `boost::multi_index_container`.
+
+- With these PR's changes, we can now "guarantee" at least 1 ancestor package worth of orphan resolution per peer at a
+  time. A fuzz test, `txorphan_protected`, and a few functional tests check this. Also, if most peers are not
+providing orphans, the unused space allocated to them can be used by more useful peers.
+
+- An earlier version of this PR implemented the new eviction strategy using the existing `TxOrphanage` data structures.
+  You can find that version of the PR [here](https://github.com/glozow/bitcoin/tree/2025-05-copy-31829).
```

Comment:
A brief summary (or a link pointing to it) of why it was changed would be helpful here, otherwise readers won't really know what to do with this I think. It's a non-trivial diff to review.

Reply:
added a couple sentences